src/rpc/methods/wallet.rs

```diff
@@ -223,6 +225,50 @@ impl RpcMethod<2> for WalletSign {
     }
 }
 
+pub enum WalletSignMessage {}
+impl RpcMethod<2> for WalletSignMessage {
+    const NAME: &'static str = "Filecoin.WalletSignMessage";
+    const PARAM_NAMES: [&'static str; 2] = ["address", "message"];
+    const API_PATHS: ApiPaths = ApiPaths::V0;
+    const PERMISSION: Permission = Permission::Sign;
+
+    type Params = (Address, Message);
+    type Ok = SignedMessage;
+
+    async fn handle(
+        ctx: Ctx<impl Blockstore + Send + Sync + 'static>,
+        (address, message): Self::Params,
+    ) -> Result<Self::Ok, ServerError> {
+        let ts = ctx.chain_store.heaviest_tipset();
+        let key_addr = ctx
+            .state_manager
+            .resolve_to_deterministic_address(address, ts)
+            .await?;
+
+        let keystore = &mut *ctx.keystore.write().await;
+        let key = match crate::key_management::find_key(&key_addr, keystore) {
+            Ok(key) => key,
+            Err(_) => {
+                let key_info = crate::key_management::try_find(&key_addr, keystore)?;
+                Key::try_from(key_info)?
+            }
+        };
+
+        let sig = crate::key_management::sign(
+            *key.key_info.key_type(),
+            key.key_info.private_key(),
+            message.cid().unwrap().to_bytes().as_slice(),
```

Comment:
I'm surprised that you're ok with panicking in this handler

Reply:
I copy-pasted the signing code, and I didn't have the mental fortitude to fix the root cause. Hashing a message should be infallible but untangling that spaghetti will be rough.

Reply:
I think I'd prefer an `expect("<here's why this is impossible>")` or a ticket link instead :)

Reply:
Done.